### ISSUE TITLE
New feature: Hooks

### ISSUE THREAD

**Issue by ghost** [NatLibFi/record-loader#10]
# Description

Allow hooking into certain parts of execution via hooks API
# Specification

_IN PROGRESS_
## Modules config

``` javascript
{
  recordSet: fooRecordSet,
  recordStore: fooRecordStore,
  hooks: {
    someEvent: fooHook
  }
  ...
}
```
## Hook environment
- Have all the modules available for use
- Can interrupt the execution
## Hook events
- **beforeTransaction** - Called before the transaction actually starts. The context (Related records) is available in the hook

**Comment by ghost** [comment]
Depends on natlibfi/record-loader-prototypes#4.

**Comment by ghost** [comment]
Implemented in d0671f3a663a1bbb38097f068a5cb9b1d7cf84b5
